src/hooks/useArNSRecordsForWallet.tsx

```diff
@@ -0,0 +1,38 @@
+import { AoArNSNameData } from '@ar.io/sdk';
+import { useGlobalState, useWalletState } from '@src/state';
+import { UseQueryResult, useQuery } from '@tanstack/react-query';
+
+export const useArNSRecordsForWallet = (): UseQueryResult<
+  Record<string, AoArNSNameData>
+> => {
+  const [{ walletAddress }] = useWalletState();
+  const [{ arioContract }] = useGlobalState();
+  return useQuery({
+    queryKey: ['arns-records-for-wallet', walletAddress?.toString()],
+    queryFn: async () => {
+      let cursor: string | undefined = undefined;
+      const arnsRecordsForWallet: Record<string, AoArNSNameData> = {};
+      let hasMore = true;
+      while (hasMore) {
+        const {
+          items: newRecords,
+          hasMore: nextHasMore,
+          nextCursor,
+        } = await arioContract.getArNSRecordsForAddress({
+          address: walletAddress?.toString() ?? '',
```

Comment:
nit: explicitly throw if no wallet address to satisfy type (also see comment on enabling the query)